packages/engine/src/datastore/batch/message.rs

```diff
@@ -101,34 +85,35 @@ impl MessageBatch {
 
         // Perform some light bound checks
         // we can't release memory on mac because we can't resize the segment
-        if cfg!(not(target_os = "macos")) && self.memory().size > LOWER_BOUND {
+        if cfg!(not(target_os = "macos")) && batch.memory().size > LOWER_BOUND {
             let upper_bound = agent_count * UPPER_MULTIPLIER;
-            if self.memory().size > upper_bound
-                && self
+            if batch.memory().size > upper_bound
+                && batch
                     .memory()
                     .target_total_size_accommodates_data_size(upper_bound, data_len)
             {
-                self.memory_mut().resize(upper_bound)?;
-                self.memory_mut().set_data_length(data_len)?;
+                batch.memory_mut().resize(upper_bound)?;
+                batch.memory_mut().set_data_length(data_len)?;
                 // Always increment when resizing
                 metaversion_to_persist.increment();
             }
         }
 
-        let old_metadata_size = self.memory().get_metadata()?.len();
+        let old_metadata_size = self.batch.memory().get_metadata()?.len();
```

Comment:
Is there a reason we're using `self.batch` here and below instead of your `batch` variable you made?

Reply:
There were no specific reasons, no. Nice spot!